rasa/core/channels/facebook.py

```diff
@@ -45,6 +45,7 @@ def _is_audio_message(message: Dict[Text, Any]) -> bool:
             "message" in message
             and "attachments" in message["message"]
             and message["message"]["attachments"][0]["type"] == "audio"
+            and not message["message"].get("is_echo")
```

Comment:
I think adding this in the different `_is_<type>_message` checks leads to these methods having more than 1 purpose. I suggest instead extracting this boolean variable in a separate function `_is_echo_message` and then adding this check once only at the top of the `message` method before all the other checks:
```
    async def message(
        self, message: Dict[Text, Any], metadata: Optional[Dict[Text, Any]]
    ) -> None:
        """Handle an incoming event from the fb webhook."""
        if self._is_echo_message(message):
                logger.warning("Received a message echo from Facebook which will be ignored.")
                return None
        # quick reply and user message both share 'text' attribute
        # so quick reply should be checked first
        if self._is_quick_reply_message(message):
              ...
```